keras_nlp/tokenizers/byte_tokenizer.py

```diff
@@ -0,0 +1,197 @@
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Byte Tokenizer."""
+
+from typing import Any
+from typing import Dict
+from typing import List
+
+import numpy as np
+import tensorflow as tf
+import tensorflow_text as tf_text
+
+from keras_nlp.tokenizers import tokenizer
+
+
+class ByteTokenizer(tokenizer.Tokenizer):
+    """Raw byte tokenizer.
+
+    This tokenizer is a vocabulary-free tokenizer which will tokenize text as
+    as raw bytes from [0, 256).
+
+    By default, the layer will output a `tf.RaggedTensor` where the last
+    dimension of the output is ragged. If `sequence_length` is set, the layer
+    will output a dense `tf.Tensor` where all inputs have been padded or
+    truncated to `sequence_length`. The output dtype can be controlled via the
+    `dtype` argument, which should be an integer type
+    (tf.int16, tf.int32, etc.).
+
+    Args:
+    lowercase: boolean. If True, the input text will be converted to lowercase
+        before tokenization.
+    sequence_length: int. If set, the output will be converted to a dense
+        tensor and padded/trimmed so all outputs are of sequence_length.
+    normalization_form: string. One of the following values: (None, "NFC",
+        "NFKC", "NFD", "NFKD"). If set, every UTF-8 string in the input tensor
+        text will be normalized to the given form before tokenizing.
+    errors: string. One of ("strict", "replace", "ignore"). Defaults to
+        "replace".Specifies the `detokenize()` behaviour when an invalid byte
+        sequence is encountered (same behaviour as
+        https://www.tensorflow.org/api_docs/python/tf/strings/unicode_transcode).
+    replacement_char: int. Defaults to 65533. The replacement character to use
+        when an invalid byte sequence is encountered and when `errors` is set to
+        "replace" (same behaviour as
+        https://www.tensorflow.org/api_docs/python/tf/strings/unicode_transcode).
+
+    Examples:
+
+    Ragged outputs.
+    >>> inputs = ["hello"]
+    >>> tokenizer = keras_nlp.tokenizers.ByteTokenizer()
+    >>> tokenizer(inputs)
+    <tf.RaggedTensor [[104, 101, 108, 108, 111]]>
+
+    Dense outputs.
+    >>> inputs = ["hello"]
+    >>> tokenizer = keras_nlp.tokenizers.ByteTokenizer(sequence_length=8)
+    >>> tokenizer(inputs)
+    <tf.Tensor: shape=(1, 8), dtype=int32,
+                numpy=array([[104, 101, 108, 108, 111,   0,   0,   0]])>
+
+    Detokenization.
+    >>> inputs = ["hello"]
+    >>> tokenizer = keras_nlp.tokenizers.ByteTokenizer()
+    >>> tokenizer.detokenize(tokenizer.tokenize(inputs))
+    <tf.Tensor: shape=(1,), dtype=string, numpy=array([b'hello'], dtype=object)>
+
+    Detokenization with errors = "replace", replacement_char = 65533.
```

Comment:
I don't think you are actually demonstrating a invalid byte sequence. You are just showing the that � will map back and forth which is what's expected without replacement. Example colab shows and actual invalid sequence.